feat: core networks for osnap

Reinis-FRP opened the pull request:
## Add a feature

### Feature Request

Add support of UMA oSnap modules on Core networks (https://coredao.org/)

### Implementation

Add support by deploying module proxy factory and oSnap module mastercopy on Core mainnet/testnet

### Additional Context

- SingletonFactory deployed at `0xce0042B868300000d44A59004Da54A005ffdcf9f` for [mainnet](https://scan.coredao.org/address/0xce0042B868300000d44A59004Da54A005ffdcf9f)/[testnet](https://scan.test.btcs.network/address/0xce0042b868300000d44a59004da54a005ffdcf9f)
- ModuleProxyFactory deployed at `0x000000000000aDdB49795b0f9bA5BC298cDda236` for [mainnet](https://scan.coredao.org/address/0x000000000000aDdB49795b0f9bA5BC298cDda236)/[testnet](https://scan.test.btcs.network/address/0x000000000000aDdB49795b0f9bA5BC298cDda236)
- OptimisticGovernor mastercopy deployed at `0x596Fd6A5A185c67aBD1c845b39f593fBA9C233aa` for [mainnet](https://scan.coredao.org/address/0x596Fd6A5A185c67aBD1c845b39f593fBA9C233aa)
- OptimisticGovernor mastercopy deployed at `0xD43463Fadd73373bE260b67F5825274F4403dAF0` for [testnet](https://scan.test.btcs.network/address/0xD43463Fadd73373bE260b67F5825274F4403dAF0)

Reinis-FRP commented:
I have read the CLA Document and I hereby sign the CLA

auryn-macmillan commented:
Thanks for the PR, @Reinis-FRP. Could you please also add Core Network and Core Network Testnet to the list of networks in `hardhat.config.ts` and confirm that the `deploy` script listed in `package.json` works on Core Network and Core Network Testnet.

Reinis-FRP commented:
> Thanks for the PR, @Reinis-FRP. Could you please also add Core Network and Core Network Testnet to the list of networks in `hardhat.config.ts` and confirm that the `deploy` script listed in `package.json` works on Core Network and Core Network Testnet?

Sure, I added config to `hardhat.config.ts` and confirmed the `deploy` script works. Though I needed to fix couple of syntax errors related to this script that I added to this PR.

Also one issue with deployment was that `ROLES_V2` mastercopy deployment is reverting on Core, so `singletonFactory.deploy.staticCall` returns zero address and the script assumes `ROLES_V2` is already deployed.

Reinis-FRP commented:
The above issue with `ROLES_V2` is not a blocker for oSnap, so this can be merged. @auryn-macmillan , let me know if there are any concerns on this
